Fail to install the project

## What did I do and see
```sh
% npm ci

> @ghostery/dnr-extension@1.0.0 preinstall /Users/tsekityam/Documents/GitHub/ghostery-dnr-extension
> rimraf src/vendor src/rule_resources

sh: rimraf: command not found
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! @ghostery/dnr-extension@1.0.0 preinstall: `rimraf src/vendor src/rule_resources`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the @ghostery/dnr-extension@1.0.0 preinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/tsekityam/.npm/_logs/2021-11-04T03_39_47_804Z-debug.log
```

## Why it failed
> `"preinstall": "rimraf src/vendor src/rule_resources"`

`rimraf` is used in preinstall stage, before it was installed as `devDependencies`, so the install failed.

## What did I do next
I changed the scripts from

```
    "preinstall": "rimraf src/vendor src/rule_resources",
    "postinstall": "vendor-copy && npm run download-cosmetics",
```

to 

```
    "postinstall": "rimraf src/vendor src/rule_resources && vendor-copy && npm run download-cosmetics",
```

Now we ran `rimraf` after it was installed. So `npm ci` should work now.

However, `npm ci` still failed.

## What did I see now

```sh
% npm ci

> core-js@3.17.2 postinstall /Users/tsekityam/Documents/GitHub/ghostery-dnr-extension/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"

Debugger listening on ws://127.0.0.1:49296/95e7263e-a211-465e-abab-241e34e12222
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!

The project needs your help! Please consider supporting of core-js:
> https://opencollective.com/core-js 
> https://patreon.com/zloirock 
> https://paypal.me/zloirock 
> bitcoin: bc1qlea7544qtsmj2rayg0lthvza9fau63ux0fstcz 

Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)

Waiting for the debugger to disconnect...

> spawn-sync@1.0.15 postinstall /Users/tsekityam/Documents/GitHub/ghostery-dnr-extension/node_modules/spawn-sync
> node postinstall


> dtrace-provider@0.8.8 install /Users/tsekityam/Documents/GitHub/ghostery-dnr-extension/node_modules/dtrace-provider
> node-gyp rebuild || node suppress-error.js

  ACTION binding_gyp_ndtp_target_build_ndtp .
-----------------------------------------------------------------
Building dtrace-provider failed with exit code 1, falling back
on stub implementation.

Re-run install with V set in your environment to see the build
output, or NODE_DTRACE_PROVIDER_REQUIRE=hard to force an
installation failure.
-----------------------------------------------------------------
  TOUCH Release/obj.target/ndtp.stamp

> @ghostery/dnr-extension@1.0.0 postinstall /Users/tsekityam/Documents/GitHub/ghostery-dnr-extension
> rimraf src/vendor src/rule_resources && vendor-copy && npm run download-cosmetics

Failed to install vendor modules: [
  [Error: ENOENT: no such file or directory, stat '/Users/tsekityam/Documents/GitHub/ghostery-dnr-lists/build'] {
    errno: -2,
    code: 'ENOENT',
    syscall: 'stat',
    path: '/Users/tsekityam/Documents/GitHub/ghostery-dnr-lists/build'
  }
]
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ghostery/dnr-extension@1.0.0 postinstall: `rimraf src/vendor src/rule_resources && vendor-copy && npm run download-cosmetics`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @ghostery/dnr-extension@1.0.0 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/tsekityam/.npm/_logs/2021-11-04T03_55_07_945Z-debug.log
```

This time `vendor-copy` failed. Because we tried to copy files from another repo which is not available to public.
```
    {
      "from": "../ghostery-dnr-lists/build",
      "to": "src/rule_resources"
    }
```

## Suggestion
I understand that the list is valuable  and you decided not to public it. But I would like to see a dummy repo that contains a few dummy records, such that others can help on this project.

--- a/.gitignore
+++ b/.gitignore
@@ -1,6 +1,7 @@
 .DS_Store
 node_modules
 build
+tmp
 dist
 src/_metadata
 src/vendor

--- a/package.json
+++ b/package.json
@@ -4,8 +4,9 @@
   "type": "module",
   "scripts": {
     "download-cosmetics": "node scripts/download-cosmetics.js",
-    "preinstall": "rimraf src/vendor src/assets/rule_resources",
-    "postinstall": "vendor-copy && npm run download-cosmetics",
+    "download-dnr-lists": "node scripts/download-dnr-lists.js",
+    "preinstall": "rimraf src/vendor",
+    "postinstall": "vendor-copy && npm run download-cosmetics && npm run download-dnr-lists",
     "build": "node scripts/build.js",
     "build.chromium": "npm run build -- --target chromium",
     "build.safari": "npm run build -- --target safari",
@@ -45,10 +46,6 @@
     "sourceDir": "./dist/"
   },
   "vendorCopy": [
-    {
-      "from": "../ghostery-dnr-lists/build",
-      "to": "src/assets/rule_resources"
-    },
     {
       "from": "node_modules/@whotracksme/webextension-packages/packages",
       "to": "src/vendor/@whotracksme"

--- a/scripts/download-dnr-lists.js
+++ b/scripts/download-dnr-lists.js
@@ -1,0 +1,48 @@
+import shelljs from 'shelljs';
+import { writeFileSync } from 'fs';
+import { resolve } from 'path';
+
+const REPO_URL = 'git@github.com:ghostery/ghostery-dnr-lists.git';
+
+shelljs.rm('-rf', 'tmp');
+shelljs.rm('-rf', 'src/assets/rule_resources/*.json');
+
+function writeJSONPlaceholder(path, content) {
+  writeFileSync(
+    resolve('src/assets/rule_resources/', path),
+    JSON.stringify(content),
+  );
+}
+
+try {
+  const result = shelljs.exec(`git clone ${REPO_URL} tmp/dnr-lists`);
+  if (result.code !== 0) {
+    throw Error("Couldn't clone repo");
+  }
+
+  shelljs.exec('cd tmp/dnr-lists && npm ci && npm start');
+
+  shelljs.cp('-R', 'tmp/dnr-lists/build/*.json', 'src/assets/rule_resources/');
+} catch (e) {
+  console.log('Generating placeholder DNR lists');
+
+  writeJSONPlaceholder('bugs.json', {
+    patterns: { host_path: {} },
+  });
+
+  writeJSONPlaceholder('categories.json', {});
+  writeJSONPlaceholder('trackers.json', {});
+  writeJSONPlaceholder('tracker_domains.json', {});
+
+  writeJSONPlaceholder('sites.json', []);
+
+  writeJSONPlaceholder('dnr-ads-network.json', []);
+  writeJSONPlaceholder('dnr-annoyances-network.json', []);
+  writeJSONPlaceholder('dnr-tracking-network.json', []);
+
+  writeJSONPlaceholder('dnr-safari-ads-network.json', []);
+  writeJSONPlaceholder('dnr-safari-annoyances-network.json', []);
+  writeJSONPlaceholder('dnr-safari-tracking-network.json', []);
+} finally {
+  shelljs.rm('-rf', 'tmp');
+}

